src/internal/observable/dom/animationFrames.ts

```diff
@@ -77,20 +77,23 @@ export interface TimestampProvider {
  *
  * @param timestampProvider An object with a `now` method that provides a numeric timestamp
  */
-export function animationFrames(timestampProvider: TimestampProvider = Date) {
+export function animationFrames(timestampProvider?: TimestampProvider) {
   return timestampProvider === Date ? DEFAULT_ANIMATION_FRAMES : animationFramesFactory(timestampProvider);
 }
 
 /**
  * Does the work of creating the observable for `animationFrames`.
  * @param timestampProvider The timestamp provider to use to create the observable
  */
-function animationFramesFactory(timestampProvider: TimestampProvider) {
+function animationFramesFactory(timestampProvider?: TimestampProvider) {
   return new Observable<number>(subscriber => {
     let id: number;
-    const start = timestampProvider.now();
-    const run = () => {
-      subscriber.next(timestampProvider.now() - start);
+    let start: DOMHighResTimeStamp | number | null = null;
+    const run = (timestamp: DOMHighResTimeStamp | number) => {
+      if (start === null) {
+        start = timestampProvider ? timestampProvider.now() : timestamp;
+      }
+      subscriber.next((timestampProvider ? timestampProvider.now() : timestamp) - start);
```

Comment:
If the timestamps provided by the API are being used, IMO they should be passed through without modification.

That is, the `start` should only be subtracted when using a caller-specified timestamp provider.

AFAICT, the point of using the API-provider timestamp is to synchronize animations and if an elapsed time is calculated from the `start` time, synchronization would only be possible between animations that started on the same animation frame. This seems like an unnecessary restriction, to me.

@benlesh What is your opinion on this? When should the operator emit an elapsed time: always, never or only if a caller-specified timestamp provider is specified?

Reply:
I see. So you'd have users calculate the elapsed time themselves?

The problem I see is they would then need to capture the initial time themselves. Maybe we can provide both?  `{ timestamp: number, elapsed: number }`

Reply:
Then again, if they already have the timestamp provider, it's trivial to get the current timestamp within whatever operation you're performing. But less trivial to know the recipe to get the elapsed time, which is more useful in animations.

Reply:
I think providing both - `{ timestamp: number, elapsed: number }` - would be the best. It'll cover all bases and is essentially self-documenting. Thanks.